pkg/util/http.go

```diff
@@ -19,6 +20,22 @@ import (
 
 const messageSizeLargerErrFmt = "received message larger than max (%d vs %d)"
 
+// BasicAuth configures basic authentication for HTTP clients.
+type BasicAuth struct {
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+}
+
+func (b *BasicAuth) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
+	f.StringVar(&b.Username, prefix+"username", "", "Username that will be set basic authentication in requests.")
+	f.StringVar(&b.Password, prefix+"password", "", "Password that will be set for basic authentication in requests.")
+}
+
+// IsZero returns false if basic authentication isn't enabled.
+func (b BasicAuth) IsZero() bool {
```

Comment:
What do you think to reverse it to `IsEnabled()`? Looks more clear.

Reply:
Agreed, I'll rename it and change the boolean logic at the site of the caller.